File: back/CommentService.Application/DTO/CommentDto.cs
namespace CommentService.Application.DTO
{
    public class CommentDto
    {
        public Guid Id { get; set; }
        public Guid PostId { get; set; }
        public Guid? ParentId { get; set; }
        public string Text { get; set; } = null!;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
        public CommentUserDto User { get; set; } = null!;
        public List<CommentMediaDto> Media { get; set; } = [];
        public CommentReactionSummaryDto Reactions { get; set; } = new();
        public int RepliesCount { get; set; }
    }
}

File: front/Сonsumer/src/interface/DTO/comments/CommentDto.ts
import type { CommentMediaDto } from "./CommentMediaDto";
import type { CommentReactionSummaryDto } from "./CommentReactionSummaryDto";
import type { CommentUserDto } from "./CommentUserDto";

export interface CommentDto {
  id: string;
  postId: string;

  parentId: string | null;

  text: string;

  createdAt: string;
  updatedAt?: string | null;

  isDeleted: boolean;

  user: CommentUserDto;

  media: CommentMediaDto[];

  reactions: CommentReactionSummaryDto;

  repliesCount: number;
}
